lib/buildTrunkPackage.nix

```diff
@@ -0,0 +1,63 @@
+{ mkCargoDerivation
+, buildDepsOnly
+, trunk
+, wasm-bindgen-cli
+, binaryen
+, nodePackages
+}:
+
+{
+  # this is not called cargoArtifacts to push users
+  # away from giving us useless artifacts built
+  # for native target instead of wasm32-unknown-unknown
+  cargoArtifacts ? null
+, trunkIndexPath ? "./index.html"
+, trunkExtraArgs ? ""
+, trunkExtraBuildArgs ? ""
+, ...
+}@origArgs:
+let
+  args = builtins.removeAttrs origArgs [
+    "trunkIndexPath"
+    "trunkExtraArgs"
+    "trunkExtraBuildArgs"
+  ];
+
+  incorrectTargetError = throw "cargoArtifacts must be built for target wasm32-unknown-unknown";
+  builtTargets = builtins.readDir "${cargoArtifacts}/target";
+  wasmTargetDir = builtTargets."wasm32-unknown-unknown" or incorrectTargetError;
+in
+
+assert cargoArtifacts == null || wasmTargetDir == "directory";
```

Comment:
Rather than doing this assertion I think a better alternative is to set `CARGO_BUILD_TARGET = "wasm32-unknown-unknown"` as a default (if not overwritten)? That would cover most of the common cases (esp with adding a default call to `cargoArtifacts = buildDepsOnly ...` when not set too) while avoiding IFD headaches or assumptions about the artifact format (since it can be customized)

Reply:
Yes this is much better, using this less intrusive env var never crossed my mind.